### PR TITLE
Internally retry failing Azure DevOps Pipeline requests

### PR DESCRIPTION
**Pull request by nickfla1-mxm**
## Description

Internally handle failing requests to Azure DevOps APIs by retrying the failing request.

## Type of change
- [x] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

None.

## Migration notes

No migration needed.

## Extra info

Azure's APIs might never return as what it seems to be a rate-limiting / anti-DDOS mechanism. 
I could observe this behavior only when Airbyte was deployed on Kubernetes, never locally. It was preventing our sources to be imported 100% of the cases.

Users who are not experiencing this issue _should not_ have any impact on their imports.

This PR introduces a new dependency.

### CODE REVIEW

File: sources/azurepipeline-source/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "commander": "^9.3.0",
     "faros-airbyte-cdk": "0.0.1",
     "nock": "^13.1.0",
+    "retry-axios": "2.6.0",
```
Comment (nickfla1-mxm) [inline sources/azurepipeline-source/package.json:38]: `retry-axios` migrates to ESM in [version 3.0.0](https://github.com/JustinBeckwith/retry-axios/releases/tag/v3.0.0), making it incompatible with the current build process. Let me know if I have missed something but I couldn't make it work with newer versions

File: sources/azurepipeline-source/resources/spec.json [hunk 2]
```diff
@@ -60,6 +60,20 @@
...
+        "title": "API Error Retries",
+        "description": "Number of times failed API calls will be retried.",
+        "default": 3
+      },
+      "api_retry_delay": {
+        "order": 7,
+        "type": "integer",
+        "title": "API Retry Delay",
+        "description": "Delay (in milliseconds) to use when retryng API calls.",
+        "default": 1000
```
Comment (nickfla1-mxm) [inline sources/azurepipeline-source/resources/spec.json:76]: These default values are what seems to be working for us

**Comment by nickfla1-mxm** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by nickfla1-mxm** [comment]
recheck

**Comment by ypc-faros** [comment]
Hi @nickfla1-mxm 👋 
Thank you for your contribution!

We have a [makeAxiosInstanceWithRetry](https://github.com/faros-ai/faros-js-client/blob/cb910ff9bd087e6ea0266ccb04a8f7f4ff041ff7/src/axios.ts#L29) that you could use here.

You can refer to this [PR](https://github.com/faros-ai/airbyte-connectors/pull/1368) where I do the same change for another source.

If possible, please rename the spec field to `max_retries`. We're trying to be consistent across sources.

Let us know if you have any questions. Thanks again 🙏 !

**Comment by nickfla1-mxm** [comment]
> Hi @nickfla1-mxm 👋 Thank you for your contribution!
> 
> We have a [makeAxiosInstanceWithRetry](https://github.com/faros-ai/faros-js-client/blob/cb910ff9bd087e6ea0266ccb04a8f7f4ff041ff7/src/axios.ts#L29) that you could use here.
> 
> You can refer to this [PR](https://github.com/faros-ai/airbyte-connectors/pull/1368) where I do the same change for another source.
> 
> If possible, please rename the spec field to `max_retries`. We're trying to be consistent across sources.
> 
> Let us know if you have any questions. Thanks again 🙏 !

Thank you for the tip! I'll update the PR soon! 🙏

File: package-lock.json
Comment (nickfla1-mxm) [inline package-lock.json:1]: I didn't expect so many changes here, `lerna info`:
```
 Environment info:

  System:
    OS: Linux 5.10 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (2) x64 Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz
  Binaries:
    Node: 18.17.0 - ~/.nvm/versions/node/v18.17.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.17.0/bin/yarn
    npm: 9.6.7 - ~/.nvm/versions/node/v18.17.0/bin/npm
  Utilities:
    Git: 2.30.2 - /usr/bin/git
  npmPackages:
    lerna: ^6.6.2 => 6.6.2 
```

File: sources/azurepipeline-source/src/azurepipeline.ts [hunk 3]
```diff
@@ -64,29 +72,39 @@ export class AzurePipeline {
...
+        params: {
+          'api-version': version,
+        },
+        headers: {
+          Authorization: `Basic ${accessToken}`,
+        },
       },
-      headers: {
-        Authorization: `Basic ${accessToken}`,
+      logger.asPino(),
```
Comment (ypc-faros) [inline sources/azurepipeline-source/src/azurepipeline.ts:87]: logger.asPino() => logger?.asPino()

**Comment by ypc-faros** [comment]
@nickfla1-mxm I have a small [comment](https://github.com/faros-ai/airbyte-connectors/pull/1374#discussion_r1537924537). Otherwise LGTM 🎉 !

Can you also please [sign](https://webdevstudios.com/2020/05/26/retroactively-sign-git-commits/) your commits?

**Comment by nickfla1-mxm** [comment]
Let me know If I can/have to do anything for the failing check, I should have [signed](https://github.com/faros-ai/airbyte-connectors/pull/1374#issuecomment-2012075431) the CLA

**Review by ypc-faros — APPROVED** [review]

**Review by tovbinm — APPROVED** [review]

**Comment by ypc-faros** [comment]
Thank you @nickfla1-mxm 🙏 !

**Pull request merged**